pcm-core.cpp

```diff
@@ -290,16 +291,17 @@ int main(int argc, char * argv[])
 	int calibrated = PCM_CALIBRATION_INTERVAL - 2; // keeps track is the clock calibration needed
 	unsigned int numberOfIterations = 0; // number of iterations
 	string program = string(argv[0]);
-	EventSelectRegister regs[4];
+	EventSelectRegister *regs;
 	PCM::ExtendedCustomCoreEventDescription conf;
 	bool show_partial_core_output = false;
 	std::bitset<MAX_CORES> ycores;
 
+        PCM * m = PCM::getInstance();
+
+        regs = (EventSelectRegister*)malloc(sizeof(EventSelectRegister) * m->getMaxCustomCoreEvents());
```

Comment:
the memory is not deallocated (which is not nice coding practice and generally is memory leak but maybe not here). I would suggest to replace 4 with PCM_MAX_CUSTOM_COUNTERS in EventSelectRegister regs[4];).

Reply:
Thank you for pinpointing this. Fixed.